sw/device/tests/otbn_randomness_test.c

```diff
@@ -27,6 +31,121 @@ static const otbn_addr_t kVarUrndOut = OTBN_ADDR_T_INIT(randomness, urnd_out);
+
+/**
+ * The otbn context handler.
+ */
+static otbn_t otbn_ctx;
+
+/**
+ * The peripheral which fired the irq to be filled by the irq handler.
+ */
+static volatile top_earlgrey_plic_peripheral_t plic_peripheral;
+
+/**
+ * The irq id to be filled by the irq handler.
+ */
+static volatile dif_rv_plic_irq_id_t irq_id;
+
+/**
+ * The otbn irq to be filled by the irq handler.
+ */
+static volatile dif_otbn_irq_t irq;
+
+/**
+ * Provides external IRQ handling for otbn tests.
+ *
+ * This function overrides the default OTTF external ISR.
+ *
+ * It performs the following:
+ * 1. Claims the IRQ fired (finds PLIC IRQ index).
+ * 2. Compute the OTBN peripheral.
+ * 3. Compute the otbn irq.
+ * 4. Clears the IRQ at the peripheral.
+ * 5. Completes the IRQ service at PLIC.
+ */
+void ottf_external_isr(void) {
+  LOG_INFO("isr");
+  CHECK_DIF_OK(dif_rv_plic_irq_claim(&plic, kTopEarlgreyPlicTargetIbex0,
+                                     (dif_rv_plic_irq_id_t *)&irq_id));
+
+  plic_peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[irq_id];
+
+  irq = (dif_otbn_irq_t)(irq_id -
+                         (dif_rv_plic_irq_id_t)kTopEarlgreyPlicIrqIdOtbnDone);
+
+  // Otbn clock is disable, so we can not acknowledge the irq. Disabling it to
+  // avoid a infinite loop here.
+  irq_global_ctrl(false);
+
+  // Complete the IRQ by writing the IRQ source to the Ibex register.
+  CHECK_DIF_OK(
+      dif_rv_plic_irq_complete(&plic, kTopEarlgreyPlicTargetIbex0, irq_id));
+}
+
+static void otbn_wait_for_done_irq(otbn_t *otbn_ctx) {
+  // Clear the otbn irq variable: we'll set it in the interrupt handler when
+  // we see the Done interrupt fire.
+  irq = irq_id = plic_peripheral = UINT32_MAX;
+  CHECK_DIF_OK(dif_otbn_irq_set_enabled(&otbn_ctx->dif, kDifOtbnIrqDone,
+                                        kDifToggleEnabled));
+
+  // OTBN should be running. Wait for an interrupt that says
+  // it's done.
+  for (;;) {
+    // This looks a bit odd, but is needed to avoid a race condition where the
+    // OTBN interrupt comes in after we load the otbn_finished flag but before
+    // we run the WFI instruction. The trick is that WFI returns when an
+    // interrupt comes in even if interrupts are globally disabled, which means
+    // that the WFI can actually sit *inside* the critical section.
+    irq_global_ctrl(false);
+    if (plic_peripheral != UINT32_MAX)
+      break;
+
+    wait_for_interrupt();
+
+    irq_global_ctrl(true);
+  }
+
+  CHECK(plic_peripheral == kTopEarlgreyPlicPeripheralOtbn,
+        "Interrupt from incorrect peripheral: (exp: %d, obs: %s)",
+        kTopEarlgreyPlicPeripheralOtbn, plic_peripheral);
+
+  // Check this is the interrupt we expected.
+  CHECK(irq_id == kTopEarlgreyPlicIrqIdOtbnDone);
+}
+
+static void otbn_init_irq() {
+  mmio_region_t plic_base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR);
+  // Initialize PLIC and configure OTBN interrupt.
```

Comment:
I think this comment doesn't actually explain that anything in particular is happening, that isn't self evident from the code, so I'd drop it.